ext/nio4r/org/nio4r/Nio4r.java

```diff
@@ -353,6 +353,16 @@ public void setSelectionKey(SelectionKey key) {
             key.attach(this);
         }
 
+        @JRubyMethod(name = "interests=")
+        public IRubyObject setInterests(ThreadContext context, IRubyObject interests){
+            if(this.isClosed(context).equals(context.getRuntime().getFalse())){
+                this.interests = interests;
```

Comment:
Similar problem here as with `ev_io_set`: this is just changing a stored property, but to actually change the behavior you need to call into the Java NIO subsystem.

Have a look at the `interestOps()` method:

https://docs.oracle.com/javase/7/docs/api/java/nio/channels/SelectionKey.html#interestOps(int)